docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1232 @@
+For example, a ``CHANNEL`` type parameter can point to any channel in a guild,
+but can be narrowed down to a specific set of channels:
+
+.. code-block:: python
+
+    from typing import Union
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: discord.abc.GuildChannel):
+        # Everything except threads
+        pass
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: discord.TextChannel):
+        # Only text channels
+        pass
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: Union[discord.Thread, discord.VoiceChannel]):
+        # Threads and voice channels only
+        pass
+
+.. note::
+
+    Union types can't mix Discord types.
+
+    Something like ``Union[discord.Member, discord.TextChannel]`` isn't possible.
+
+Refer to the :ref:`type conversion table <type_conversion>` for full information on sub-types.
+
+Describing
++++++++++++
+
+Descriptions are added to parameters using the :func:`.app_commands.describe` decorator,
+where each keyword is treated as a parameter name.
+
+.. code-block:: python
+    :emphasize-lines: 2-5
+
+    @client.tree.command()
+    @app_commands.describe(
+        content="the text to repeat",
+        n_times="the number of times to repeat the text"
+    )
+    async def repeat(interaction: discord.Interaction, content: str, n_times: int):
+        to_send = textwrap.shorten(f"{content} " * n_times, width=2000)
+        await interaction.response.send_message(to_send)
+
+These show up on Discord just beside the parameter's name:
+
+.. image:: /images/guide/app_commands/repeat_command_described.png
+
+In addition to the decorator, parameter descriptions can also be added using
+Google, Sphinx or Numpy style docstrings.
+
+Examples using a command to add 2 numbers together:
+
+.. code-block:: python
+
+    @client.tree.command() # numpy
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        Parameters
+        -----------
+        a: int
+            left operand
+        b: int
+            right operand
+        """
+
+        await interaction.response.send_message(f"{a} + {b} is {a + b}!")
+
+    @client.tree.command() # google
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        Args:
+            a (int): left operand
+            b (int): right operand
+        """
+
+    @client.tree.command() # sphinx
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        :param a: left operand
+        :param b: right operand
+        """
+
+If both are used, :func:`.app_commands.describe` always takes precedence.
+
+Naming
+^^^^^^^
+
+Since parameter names are confined to the rules of Python's syntax,
```

Comment:
Slash command parameter also have confined rules, I feel like they should be mentioned in here too.
https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-naming